lineapy/graph_reader/session_artifacts.py

```diff
@@ -264,24 +264,40 @@ def _update_node_context(self):
                         len(self.node_context[node_id].dependent_variables)
                         == 0
                     )
-                    has_assigned_before = (
-                        self.input_parameters_node.get(var, None) is not None
+                    previous_assignment = self.input_parameters_node.get(
+                        var, None
                     )
-                    if is_literal_assignment:
-                        if has_assigned_before:
+
+                    if previous_assignment is None:
+                        self.input_parameters_node[var] = node_id
+                    else:
+                        previous_assignment_is_literal = (
+                            len(
+                                self.node_context[
+                                    previous_assignment
+                                ].dependent_variables
```

Comment:
dont know enough to evaluate what dependen_variables would do. Eg x = LineaRegression() has no depedent_variables i think but its not literal. prefer to be more explicit here and do isinstance(x, (int,bool,str,float...))

Reply:
Question for @mingjerli. This was the original check written in the code for the happy path as well. I'm not actually aware why this is the exact check we are looking for (probably that it doesnt require slicing out another node collection for the parameter) and why we are calling a literal.

Reply:
I can address this in a separate PR.